### PR TITLE
Cleaner logs

### PR DESCRIPTION
**Pull request by arvindsv**
Reduce logging. Stacktraces only in DEBUG mode, etc.

### CODE REVIEW

File: gocd-filebased-authentication-plugin/build.gradle [hunk 1]
```diff
@@ -15,11 +15,12 @@
  */
 
 dependencies {
-  compileOnly group: 'cd.go.plugin', name: 'go-plugin-api', version: '17.2.0'
+  compileOnly group: 'cd.go.plugin', name: 'go-plugin-api', version: '18.1.0'
   compile group: 'com.google.code.gson', name: 'gson', version: '2.8.0'
   compile project(':bcrypt')
+  compile group: 'javax.xml.bind', name: 'jaxb-api', version: '2.3.0'
```
Comment (ketan) [inline gocd-filebased-authentication-plugin/build.gradle]: Why was this required?

Comment (arvindsv) [reply gocd-filebased-authentication-plugin/build.gradle]: @ketan To compile with JDK 9 and above. They removed some stuff in it:

http://mail.openjdk.java.net/pipermail/jdk9-dev/2016-May/004309.html
https://stackoverflow.com/questions/43574426/how-to-resolve-java-lang-noclassdeffounderror-javax-xml-bind-jaxbexception-in-j

We've done this before, in another plugin too: https://github.com/gocd-contrib/go-nuget-poller-plugin-2.0/commit/9b8bd794af54f202efd7aa9defc9a6583637dccd#diff-c197962302397baf3a4cc36463dce5eaL46

Comment (arvindsv) [reply gocd-filebased-authentication-plugin/build.gradle]: Here's an example of the usage in this code: https://github.com/gocd/gocd-filebased-authentication-plugin/blob/672cb0f31ff6fda25cf9c3a1babb07aab45d52fd/gocdpasswd/src/main/java/cd/go/authentication/crypt/hash/PBKDF2Provider.java#L23

**Comment by arvindsv** [comment]
Please don't merge till @ankitsri11 finishes testing it.

**Comment by arvindsv** [comment]
@ankitsri11 found that this exception is still thrown (see below). We will be working together on fixing that:

```
2019-01-14 17:40:41,419 ERROR [qtp632249781-30] PasswordFilePlugin:128 - [Authenticate] Failed to authenticate user: `john`
cd.go.authentication.passwordfile.exception.AuthenticationException: Unable to authenticate user, user not found or bad credentials
    at cd.go.authentication.passwordfile.Authenticator.authenticate(Authenticator.java:73)
    at cd.go.authentication.passwordfile.executor.UserAuthenticationExecutor.execute(UserAuthenticationExecutor.java:56)
    at cd.go.authentication.passwordfile.PasswordFilePlugin.handle(PasswordFilePlugin.java:68)
    at com.thoughtworks.go.plugin.infra.DefaultPluginManager.lambda$submitTo$0(DefaultPluginManager.java:143)
    at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework.executeActionOnTheService(FelixGoPluginOSGiFramework.java:225)
    at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework.doOn(FelixGoPluginOSGiFramework.java:202)
    at com.thoughtworks.go.plugin.infra.DefaultPluginManager.submitTo(DefaultPluginManager.java:140)
    at com.thoughtworks.go.plugin.access.PluginRequestHelper.submitRequest(PluginRequestHelper.java:48)
    at com.thoughtworks.go.plugin.access.authorization.AuthorizationExtension.authenticateUser(AuthorizationExtension.java:118)
    at com.thoughtworks.go.server.newsecurity.providers.PasswordBasedPluginAuthenticationProvider.authenticateWithExtension(PasswordBasedPluginAuthenticationProvider.java:78)
    at com.thoughtworks.go.server.newsecurity.providers.PasswordBasedPluginAuthenticationProvider.authenticateWithExtension(PasswordBasedPluginAuthenticationProvider.java:39)
    at com.thoughtworks.go.server.newsecurity.providers.AbstractPluginAuthenticationProvider.authenticateUser(AbstractPluginAuthenticationProvider.java:117)
    at com.thoughtworks.go.server.newsecurity.providers.AbstractPluginAuthenticationProvider.authenticate(AbstractPluginAuthenticationProvider.java:84)
    at com.thoughtworks.go.server.newsecurity.controllers.AuthenticationController.performLogin(AuthenticationController.java:92)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```

**Comment by ankitsri11** [comment]
@arvindsv now it doesn't log any login failure INFO message as it gets logged in LDAP authentication plugin. Not even for an invalid account (user not available in either of the plugins).

**Comment by arvindsv** [comment]
:) Ha. Yes, that's right. How about now?

**Comment by arvindsv** [comment]
@ketan took a look too. Merging.
